One transaction per request

BacLuc opened the pull request:
replaces #2391

carlobeltrame commented on api/src/DataPersister/CategoryDataPersister.php:
```diff
@@ -34,4 +36,10 @@ public function beforeCreate($data): BaseEntity {
         return $data;
     }
+
+    public function beforeRemove($data): ?BaseEntity {
+        $this->validator->validate($data, ['groups' => ['delete']]);
+
+        return null;
+    }
```
Wollen wir das nicht mal in den AbstractDataPersister hochnehmen? Dann könnte ich bei mehr Entities einfach jederzeit beginnen, Delete Validations zu schreiben. Und vielleicht auch gleich für alle Entities einen trivialen DataPersister anlegen, sodass ich das bei allen Entities kann?

BacLuc replied:
Mache ich gerne, aber in einem neuen PR.

BacLuc commented:
@usu hatte recht, flush braucht es nicht, und wenn man kein flush macht, braucht es auch kein cascade remove.
Ich würde die Files so lassen (falls sie so gut sind), und noch die commits herumschieben

BacLuc committed:
DeleteCategoryTest: delete category with no activities

Because it should not be able to delete a category with activities.

BacLuc committed:
Category: validate that category has no activities on delete

BacLuc committed:
api: run requests in transaction

That we have an all or nothing semantic.

BacLuc committed:
api: test OPTIONS request against root endpoint in RootTest.

Because there api platform takes a shortcut and stops the propagation
of the request event after priority 250.
